src/fides/api/alembic/migrations/versions/1e750936a141_migrate_to_fideslang_2_0_uses_and_.py

```diff
@@ -0,0 +1,289 @@
+            )
+
+        # Update the collections objects
+        collections: str = json.dumps(row["collections"])
+
+        for key, value in data_label_map.items():
+            collections = collections.replace(key, value)
+
+        update_collections_query: TextClause = text(
+            "UPDATE ctl_datasets SET collections = :updated_collections WHERE id= :dataset_id"
+        )
+        bind.execute(
+            update_collections_query,
+            {"dataset_id": row["id"], "updated_collections": collections},
+        )
+
+
+def update_system_ingress_egress_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade system DataFlow objects"""
+    existing_systems: ResultProxy = bind.execute(
+        text("SELECT id, egress, ingress FROM ctl_systems;")
+    )
+
+    for row in existing_systems:
+        ingress = row["ingress"]
+        egress = row["egress"]
+
+        # Do a blunt find/replace
+        if ingress:
+            for item in ingress:
+                item["data_categories"] = [
+                    data_label_map.get(label, label)
+                    for label in item.get("data_categories", [])
+                ]
+
+        if egress:
+            for item in egress:
+                item["data_categories"] = [
+                    data_label_map.get(label, label)
+                    for label in item.get("data_categories", [])
+                ]
+
+        if ingress:
+            update_ingress_query: TextClause = text(
+                "UPDATE ctl_systems SET ingress = :updated_ingress WHERE id= :system_id"
+            )
+            bind.execute(
+                update_ingress_query,
+                {"system_id": row["id"], "updated_ingress": json.dumps(ingress)},
+            )
+
+        if egress:
+            update_egress_query: TextClause = text(
+                "UPDATE ctl_systems SET egress = :updated_egress WHERE id= :system_id"
+            )
+            bind.execute(
+                update_egress_query,
+                {"system_id": row["id"], "updated_egress": json.dumps(egress)},
+            )
+
+
+def update_privacy_declaration_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade data uses from fideslang 1.4 for privacy declarations"""
+    existing_ctl_policies: ResultProxy = bind.execute(
+        text("SELECT id, data_categories FROM privacydeclaration;")
+    )
+    for row in existing_ctl_policies:
+        labels: List[str] = [
+            data_label_map.get(label, label) for label in row["data_categories"]
+        ]
+
+        update_label_query: TextClause = text(
+            "UPDATE privacydeclaration SET data_categories = :updated_label WHERE id= :declaration_id"
+        )
+        bind.execute(
+            update_label_query,
+            {"declaration_id": row["id"], "updated_label": labels},
+        )
+
+
+def update_ctl_policy_data_categories(
```

Comment:
There's a new field `PrivacyDeclaration.shared_categories` going out in the same release that is also a data category field- I assume only staging environments would have a potentially old value populated, so maybe not worth migrating but just pointing this out.

Reply:
thank you for calling it out!